sdk/test/Services/S3/UnitTests/Custom/UseArnRegionTests.cs

```diff
@@ -39,18 +39,49 @@ namespace AWSSDK.UnitTests
-        private static readonly string ProfileText = @"[default]
-                                                         region=us-west-2
-                                                         aws_access_key_id=default_aws_access_key_id
-                                                         aws_secret_access_key=default_aws_secret_access_key
-                                                         s3_use_arn_region=true
-                                                         [other]
-                                                         region=us-west-2
-                                                         aws_access_key_id=other_aws_access_key_id
-                                                         aws_secret_access_key=other_aws_secret_access_key
-                                                         ";
-*/
+        private const string AWS_S3_USE_ARN_REGION_ENVIRONMENT_VARIABLE = "AWS_S3_USE_ARN_REGION";
+        private const string AWS_PROFILE_ENVIRONMENT_VARIABLE = "AWS_PROFILE";
+        private const string AWS_CONFIG_ENVIRONMENT_VARIABLE = "AWS_CONFIG_FILE";
+
+        private string _beginningS3UseArnRegionEnvironmentValue;
+        private string _beginningAWSProfileEnvironmentValue;
+        private string _tempCredentialsFilePath;
+
+        private static readonly string ProfileText = new StringBuilder()
+            .AppendLine("[default]")
+            .AppendLine("region=us-west-2")
+            .AppendLine("aws_access_key_id=default_aws_access_key_id")
+            .AppendLine("aws_secret_access_key=default_aws_secret_access_key")
+            .AppendLine("[profile use-arn-region-enabled]")
+            .AppendLine("s3_use_arn_region=true")
+            .AppendLine("[profile use-arn-region-disabled]")
+            .AppendLine("s3_use_arn_region=false")
+            .ToString();
+
+        [TestInitialize]
+        public void Initialize()
+        {
+            // Save off current environment variable value to restore later
+            _beginningS3UseArnRegionEnvironmentValue = Environment.GetEnvironmentVariable(AWS_S3_USE_ARN_REGION_ENVIRONMENT_VARIABLE);
+            _beginningAWSProfileEnvironmentValue = Environment.GetEnvironmentVariable(AWS_PROFILE_ENVIRONMENT_VARIABLE);
+            // Then clear the current value so every test is starting from a clean slate
+            Environment.SetEnvironmentVariable(AWS_PROFILE_ENVIRONMENT_VARIABLE, "");
+            // set credentials file and use it to load CredentialProfileStoreChain
+
+            // set credentials file and use it to load CredentialProfileStoreChain
+            _tempCredentialsFilePath = Path.GetTempFileName();
+            File.WriteAllText(_tempCredentialsFilePath, ProfileText);
+            ReflectionHelpers.Invoke(typeof(AmazonS3Config), "credentialProfileChain", new CredentialProfileStoreChain(_tempCredentialsFilePath));
+            ReflectionHelpers.Invoke(typeof(AmazonS3Config), "_triedToResolveProfile", false);
```

Comment:
Sorry I didn't notice this during my initial review, but why is using reflection to set a private property required? Is it because you uncommented the last 2 tests in this file?

Reply:
This wasn't here in the initial review. This is something I added after removing the `AWS_CONFIG_FILE` option in the CredentialsProfileChain constructor in `AmazonS3Config`. Now that we don't have the option to specify via environment variable, we must use reflection so the `CredentialProfileChain` knows where to look

Reply:
Discussed offline, but this is the same pattern used in https://github.com/aws/aws-sdk-net/blob/68f6a0520854f18fd991fd87cfd847995b075ab3/sdk/test/Services/S3/UnitTests/Custom/DisableS3ExpressSessionAuthTests.cs (another S3 option that can be set either via environment variable or config).

We should follow the same pattern for the tests here (just verify the precedence is correct), and when support for `AWS_CONFIG_FILE` is added remove the reflection usages.